File: app/src/main/java/mobiletest/yucef/ca/mobiletest/model/ContactInfo.java
package mobiletest.yucef.ca.mobiletest.model;

public class ContactInfo {
    String website;
    String email;
    String phoneNumber;
    String faxNumber;
    String tollFree;

    public ContactInfo(String website, String email, String phoneNumber, String faxNumber, String tollFree) {
        this.website = website;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.faxNumber = faxNumber;
        this.tollFree = tollFree;
    }

    public String getWebsite() {
        return website;
    }

    public String getEmail() {
        return email;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getFaxNumber() {
        return faxNumber;
    }

    public String getTollFree() {
        return tollFree;
    }
}

File: app/src/main/java/com/quickseries/mobiletest/domain/resources/model/ContactInfo.kt
package com.quickseries.mobiletest.domain.resources.model

import com.quickseries.mobiletest.data.resources.ContactInfoResponse
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

data class ContactInfo(
    val website: String? = null,
    val email: String? = null,
    val phoneNmber: String? = null,
    val faxNumber: String? = null,
    val tollFree: String? = null
) {
    fun isEmpty() = website == null && email == null && phoneNmber == null && faxNumber == null && tollFree == null
}

suspend fun ContactInfoResponse.toContactInfo() = withContext(Dispatchers.Default) {
    ContactInfo(
        websites.firstOrNull(),
        emails.firstOrNull(),
        phoneNumbers.firstOrNull(),
        faxNumbers.firstOrNull(),
        tollFrees.firstOrNull()
    )
}